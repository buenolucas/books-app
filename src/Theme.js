import {getFontStyleObject} from './utils/font';

const Theme = {};

// ------------------------------------------------------------
// Sizing
// ------------------------------------------------------------
Theme.spacing = {
  xTiny: 4,
  tiny: 8,
  small: 16,
  base: 24,
  large: 48,
  xLarge: 64,
};

Theme.specifications = {
  headerHeight: 56,
  bookCoverAspectRation: 0.69402985075,
};
// ------------------------------------------------------------
// Colors
// ------------------------------------------------------------

const colors = {
  paleGrey: '#effaff',
  paleGreyTwo: '#f3f6fd',
  slateGray: '#595a5c',
  robinEgg: '#56c8ff',
  darkSky: '#47b7ed',
  steel: '#86878b',
  darkGrey: '#2d2d2e',
  lightPeriwinkle: '#dee1e6',
  slateGray: '#595a5c',
  ice: '#f2fdf6',
  white: '#FFFFFF',
};

Theme.gray = {
  darkest: '#121212',
  darker: '#1a1a1a',
  dark: '#353535',
  light: '#828282',
  lighter: '#cfcfcf',
  lightest: '#f8f8f8',
};

Theme.colors = {
  ...colors,
  primary: colors.robinEgg,
  primaryVariant: colors.darkSky,
  text: colors.slateGray,
  background: colors.white,
  buttonPrimary: colors.robinEgg,
  buttonDefault: colors.paleGrey,
};

// ------------------------------------------------------------
// Typography
// ------------------------------------------------------------
Theme.typography = {
  largeTitle: {
    fontSize: 22,
    ...getFontStyleObject({weight: 'Medium'}),
  },
  header: {
    fontSize: 14,
    letterSpacing: 0.5,
    ...getFontStyleObject({weight: 'Medium'}),
  },
  body: {
    fontSize: 14,
    ...getFontStyleObject({weight: 'Regular'}),
  },
  caption1: {
    fontSize: 18,
  },
  input: {
    fontSize: 15,
    ...getFontStyleObject({weight: 'Regular'}),
  },
  caption1: {
    fontSize: 15,
    ...getFontStyleObject({weight: 'Medium'}),
  },
  caption2: {
    fontSize: 14,
    ...getFontStyleObject({weight: 'Medium'}),
  },
};

export default Theme;
