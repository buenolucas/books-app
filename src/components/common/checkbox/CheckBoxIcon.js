import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const Container = styled.View`
  border: 2px solid
    ${({checked, theme}) =>
      checked ? theme.colors.robinEgg : theme.colors.lightPeriwinkle};
  width: 16px;
  height: 16px;
  border-radius: 3px;
  padding: 2px;
`;

const Filled = styled.View`
  background-color: ${({checked, theme}) =>
    checked ? theme.colors.robinEgg : theme.colors.lightPeriwinkle};
  width: 100%;
  height: 100%;
`;
const CheckBoxIcon = ({checked, onPress}) => {
  return (
    <Container checked={checked}>
      {checked && <Filled checked={checked} />}
    </Container>
  );
};
CheckBoxIcon.propTypes = {
  checked: PropTypes.bool,
};

export default CheckBoxIcon;
