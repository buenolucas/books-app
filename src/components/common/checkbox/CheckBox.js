import React from 'react';
import PropTypes from 'prop-types';
import styled, {withTheme} from 'styled-components/native';

import Typography from '../Typography';
import CheckBoxIcon from './CheckBoxIcon';

const Component = styled.TouchableOpacity`
  margin: ${({theme}) => theme.spacing.tiny}px
  margin-left: 10px;
  margin-right: 10px;

`;
const Wrapper = styled.View`
  flex-direction: row;
  align-items: center;
`;
const CheckBox = ({theme, checked, label, onPress}) => {
  const accessibilityStates = [
    ...(checked ? ['checked'] : []),
    ...(!checked ? ['unchecked'] : []),
  ];

  const textStyle = {
    marginLeft: theme.spacing.tiny,
    color: checked ? theme.colors.darkGrey : theme.colors.steel,
  };
  return (
    <Component
      onPress={onPress}
      accessibilityRole="checkbox"
      accessibilityStates={accessibilityStates}>
      <Wrapper>
        <CheckBoxIcon checked={checked} />
        <Typography type="body" style={textStyle}>
          {label}
        </Typography>
      </Wrapper>
    </Component>
  );
};

CheckBox.propTypes = {
  onPress: PropTypes.func,
  label: PropTypes.string,
  checked: PropTypes.bool,
};

CheckBox.defaultProps = {
  checked: false,
};

export default withTheme(CheckBox);
