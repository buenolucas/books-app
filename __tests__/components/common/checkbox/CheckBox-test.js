import React from 'react';
import {Image, View, Text} from 'react-native';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import Theme from '../../../../src/Theme';
import CheckBox from '../../../../src/components/common/checkbox/CheckBox';
import CheckBoxIcon from '../../../../src/components/common/checkbox/CheckBoxIcon';

describe('CheckBox', () => {
  it('should render without issues', () => {
    const component = shallow(<CheckBox theme={Theme} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
