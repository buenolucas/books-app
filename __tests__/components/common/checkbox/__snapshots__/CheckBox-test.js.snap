// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should render without issues 1`] = `
<CheckBox
  checked={false}
  theme={
    Object {
      "colors": Object {
        "background": "#FFFFFF",
        "buttonDefault": "#effaff",
        "buttonPrimary": "#56c8ff",
        "darkGrey": "#2d2d2e",
        "darkSky": "#47b7ed",
        "ice": "#f2fdf6",
        "lightPeriwinkle": "#dee1e6",
        "paleGrey": "#effaff",
        "paleGreyTwo": "#f3f6fd",
        "primary": "#56c8ff",
        "primaryVariant": "#47b7ed",
        "robinEgg": "#56c8ff",
        "slateGray": "#595a5c",
        "steel": "#86878b",
        "text": "#595a5c",
        "white": "#FFFFFF",
      },
      "gray": Object {
        "dark": "#353535",
        "darker": "#1a1a1a",
        "darkest": "#121212",
        "light": "#828282",
        "lighter": "#cfcfcf",
        "lightest": "#f8f8f8",
      },
      "spacing": Object {
        "base": 24,
        "large": 48,
        "small": 16,
        "tiny": 8,
        "xLarge": 64,
        "xTiny": 4,
      },
      "specifications": Object {
        "bookCoverAspectRation": 0.69402985075,
        "headerHeight": 56,
      },
      "typography": Object {
        "body": Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
        },
        "caption1": Object {
          "fontFamily": "Roboto",
          "fontSize": 15,
          "fontStyle": "normal",
          "fontWeight": "500",
        },
        "caption2": Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "500",
        },
        "header": Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "500",
          "letterSpacing": 0.5,
        },
        "input": Object {
          "fontFamily": "Roboto",
          "fontSize": 15,
          "fontStyle": "normal",
          "fontWeight": "400",
        },
        "largeTitle": Object {
          "fontFamily": "Roboto",
          "fontSize": 22,
          "fontStyle": "normal",
          "fontWeight": "500",
        },
        "title": Object {
          "fontFamily": "Roboto",
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": "500",
        },
      },
    }
  }
/>
`;
